qiskit/visualization/pulse_v2/core.py

```diff
@@ -0,0 +1,493 @@
+        if backend is not None:
+            self._load_iqx_backend(backend)
+
+        # overwrite default values
+        if drive_los is not None:
+            self.d_los.update(drive_los)
+
+        if control_los is not None:
+            self.c_los.update(control_los)
+
+        if measure_los is not None:
+            self.m_los.update(measure_los)
+
+        if dt is not None:
+            self.dt = dt
+
+    def _load_iqx_backend(self,
+                          backend: BaseBackend):
+        """A helper function to extract system property from IQX backend instance.
+
+        Notes:
+            The modulation frequencies of control channels should be defined in terms of
+            the modulation frequencies of drive channels. This is the syntax of
+            IQX backends. If the backend is provided by a third party provider,
+            this function may crash or may return wrong frequency values.
+
+        Args:
+            backend: Backend object to play the schedule.
+        """
+        configuration = backend.configuration()
+        defaults = backend.defaults()
+
+        self.dt = configuration.dt
+
+        self.d_los = dict(enumerate(defaults.qubit_freq_est))
+        self.m_los = dict(enumerate(defaults.meas_freq_est))
+        self.c_los = dict()
+
+        for ind, u_lo_mappers in enumerate(configuration.u_channel_lo):
+            temp_val = 0
+            for u_lo_mapper in u_lo_mappers:
+                temp_val = self.d_los[u_lo_mapper.q] * complex(*u_lo_mapper.scale)
+            self.c_los[ind] = temp_val.real
+
+    def load_program(self, program: Union[pulse.Waveform, pulse.ParametricPulse, pulse.Schedule]):
+        """Load a program to draw.
+
+        Args:
+            program: `Waveform`, `ParametricPulse`, or `Schedule` to draw.
+
+        Raises:
+            VisualizationError: When input program is invalid data format.
+        """
+        if isinstance(program, pulse.Schedule):
+            self._schedule_loader(program)
+        elif isinstance(program, (pulse.Waveform, pulse.ParametricPulse)):
+            self._waveform_loader(program)
+        else:
+            raise VisualizationError('Data type %s is not supported.' % type(program))
+
+    def _waveform_loader(self, program: Union[pulse.Waveform, pulse.ParametricPulse]):
+        """Load Waveform instance.
+
+        This function is sub-routine of py:method:`load_program`.
+
+        Args:
+            program: `Waveform` to draw.
+        """
+        sample_channel = types.WaveformChannel()
+        inst_tuple = types.InstructionTuple(t0=0,
+                                            dt=self.dt,
+                                            frame=types.PhaseFreqTuple(phase=0, freq=0),
+                                            inst=pulse.Play(program, sample_channel))
+
+        self.set_time_range(0, program.duration)
+
+        # generate waveform related elements
+        for gen in PULSE_STYLE['generator.waveform']:
```

Comment:
Because the stylesheet is called from everywhere, i.e. generators callbacks, layout callbacks, event manager and plotter. Sometime they are nested function call and passing the stylesheet instance to everything may make a function signature complicated. I think matplotlib has the same situation and it has rcParams as a global instance.
https://github.com/matplotlib/matplotlib/blob/5703ac49691757d4f32eb419c9e71af87586e504/lib/matplotlib/__init__.py#L911

But I didn't think about side-effects. Do you know any potential issue with it?

Reply:
Just because matplotlib does it, doesn't mean its necessarily good :). The issue is that a global singleton configuration induces tight-coupling between classes. It would be better to pass at the configuration to the required classes/functions. At the very least it should be able to pass a config, and fall back to the default if not passed. Otherwise the only way to communicate to these method's internals is through global state which is asking for a headache down the road.

Reply:
I understand your concern. Let me update the framework. I'll move this back to WIP.